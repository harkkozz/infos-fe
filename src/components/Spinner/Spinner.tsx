import React from 'react';

import { Spin } from 'antd';

type SpinSize = 'default' | 'large' | 'small';

interface Props {
  size?: SpinSize;
  isSuspense?: boolean;
}

const Spinner = ({ size = 'default', isSuspense = false }: Props) => {
  const spinSuspenseStyle: React.CSSProperties = {
    position: 'absolute',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%'
  };

  return <Spin style={isSuspense && spinSuspenseStyle} size={size} />;
};

export default Spinner;
