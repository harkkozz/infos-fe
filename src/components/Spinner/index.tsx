import React from 'react';

import { Spin } from 'antd';

type SpinSize = 'default' | 'large' | 'small';

interface Props {
  size?: SpinSize;
  isSuspense?: boolean;
}

const Spinner: React.FC<Props> = ({ size = 'default', isSuspense = false }) => {
  const spinSuspenseStyle: React.CSSProperties = isSuspense && {
    position: 'absolute',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%'
  };

  return <Spin style={{ ...spinSuspenseStyle }} size={size} />;
};

export default Spinner;
