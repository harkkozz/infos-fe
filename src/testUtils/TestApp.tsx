import React, { PropsWithChildren } from 'react';

import { MemoryRouter } from 'react-router-dom';

interface Props {
  path?: string;
}

const Test: React.FC<PropsWithChildren<Props>> = ({ path, children }) => {
  return <MemoryRouter initialEntries={[!path ? '/' : path]}>{children}</MemoryRouter>;
};

export default Test;
