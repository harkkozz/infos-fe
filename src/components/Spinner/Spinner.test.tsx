import React from 'react';

import { render } from '@testing-library/react';
import TestApp from 'testUtils/TestApp';
import { expect, it } from 'vitest';

import Spinner from 'components/Spinner';

it('should match snapshot', () => {
  const { baseElement } = render(
    <TestApp>
      <Spinner />
    </TestApp>
  );
  expect(baseElement).toMatchSnapshot();
});

it('should set isSuspense to true', () => {
  const { baseElement } = render(
    <TestApp>
      <Spinner isSuspense />
    </TestApp>
  );

  expect(baseElement).toMatchSnapshot();
});
