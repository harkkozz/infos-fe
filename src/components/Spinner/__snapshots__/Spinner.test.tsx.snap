// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<body>
  <div>
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1vtf12y"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</body>
`;

exports[`should set isSuspense to true 1`] = `
<body>
  <div>
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1vtf12y"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
  <div>
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1vtf12y"
      style="position: absolute; display: flex; justify-content: center; align-items: center; width: 100%; height: 100%;"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</body>
`;
